hashing/te-tag-query-java/com/facebook/threatexchange/IndicatorTypeFilterer.java

```diff
@@ -0,0 +1,144 @@
+// ================================================================
+// Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+// ================================================================
+
+package com.facebook.threatexchange;
+
+import java.io.PrintStream;
+import java.util.stream.Stream;
+import java.util.Map;
+import java.util.HashMap;
+
+class IndicatorTypeFilterer {
+  private String _uppercaseName;
+  private String _lowercaseName;
+
+  // The uppercase names are the ones coming back in the 'type' field in TE API
+  // requests. The lowercase names are standard shorthands.
+  //
+  // Note: this data structure Could be a hashmap-literal but the syntax gets
+  // neater in Java 9 & let's keep this code backward-compatible with various
+  // Java versions.
+  private static final String[][] NAMES = {
+    {"ADJUST_TOKEN", "adjustToken"},
+    {"API_KEY", "api_key"},
```

Comment:
In the comment, it says that lower case names are "standard shorthands"

How do you decide whether to camelCase or lower_with_underscore names?

Reply:
@Dcallies thanks! These were taking from a TE-internal hashmap. For purposes of this program, though, they're solely conveniences and there is no need to preserve compatibility with the source enum. We can change them, for sure ...